v2/contribute/decisions/passwordless/0003-claim-redirection-on-fe.mdx

```diff
@@ -0,0 +1,61 @@
+---
+id: "0003"
+title: Use configurable callbacks to redirect on claim validation success/failure
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Use configurable callbacks to redirect on claim validation success/failure
+
+<DecisionHeader status="proposed" lastUpdate="2022-11-28" created="2022-11-28" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+Apps have to handle redirection if a claim fails and also handle redirecting back if necessary. E.g.:
+- While using a custom 2FA claim, the app has to check if there is a validation error for the claim in the `invalidClaims` array and handle redirection.
+- On the 2FA page, the app has to check if that validation error disappeared (so the user successfully logged in using 2FA)
+
+## Considered Options
+
+* Recommend creating wrapper components
+* Add a callback that returns a react component to render on failure
+* Add callbacks (onFailure, onSuccess) that returns a string (or undefined) we can redirect to
+* Add a separate component that handles these checks/mapping
+
+## Decision Outcome
+
+Add callbacks (onFailure, onSuccess) that returns a string (or undefined) we can redirect to.
```

Comment:
would the string be just a path? or even full URLs? Pros / cons of that? (I ask this cause for example, in getRedirect function, we only allow return of path)